### PR TITLE
feat: Save functionality using enter

### LINKED ISSUE #373: Submit lesson on click enter
### Feature Description

In a self paced course, if the user hits enter on the new lesson modal trigger the submit functionality similar on the Save button

image

### Use Case

_No response_

### Proposed Solution

_No response_

### Alternatives (optional)

_No response_

### Additional Context

_No response_

### Please check the boxes that apply to this feature request.

- [ ] I have searched the existing feature requests to make sure this is not a duplicate.
- [ ] I have provided a detailed description of the requested feature.
- [ ] I have explained the use case or scenario for this feature.
- [ ] I have included any relevant technical details or design suggestions.
- [ ] I understand that this is a suggestion and that there is no guarantee of implementation.

### PR DESCRIPTION
**Pull request by smritidoneria**
## What does this PR do?

This pr handles the functionality of the enter key mouse event to work like submit. When the user clicks enter, the lesson title will be submitted and the dialog box will be closed.

Fixes  #373

https://github.com/rotimi-best/classroomio/assets/109029101/b75c3faa-2653-49d4-9f8d-643a7989165a

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Chore (refactoring code, technical debt, workflow improvements)
- [x] Enhancement (small improvements)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change adds a new database migration
- [ ] This change requires a documentation update

## How should this be tested?

- Test A
Go to the course
Create a lesson title and click enter, the lesson title will be submitted.
- Test B

### Required

- [x] Filled out the "How to test" section in this PR
- [x] Self-reviewed my own code
- [x] Commented on my code in hard-to-understand bits
- [ ] Ran `pnpm build`
- [x] Checked for warnings, there are none
- [x] Removed all `console.logs`
- [x] Merged the latest changes from main onto my branch with `git pull origin main`
- [x] My changes don't cause any responsiveness issues

### Appreciated

- [ ] If a UI change was made: Added a screen recording or screenshots to this PR
- [ ] Updated the ClassroomIO Docs if changes were necessary

### CODE REVIEW

File: apps/dashboard/src/lib/components/Course/components/Lesson/NewLessonModal.svelte [hunk 1]
```diff
@@ -44,19 +44,24 @@
...
+  };
 </script>
 
 <Modal
+
   onClose={() => ($handleAddLessonWidget.open = false)}
   bind:open={$handleAddLessonWidget.open}
   width="w-[80%] md:w-[65%]"
-  maxWidth="max-w-2xl"
+  maxWidth="max-w-2xl "
```
Comment (rotimi-best) [inline apps/dashboard/src/lib/components/Course/components/Lesson/NewLessonModal.svelte]: Please can you undo this change?

File: apps/dashboard/src/lib/components/Course/components/Lesson/NewLessonModal.svelte [hunk 2]
```diff
@@ -44,19 +44,24 @@
...
   containerClass="overflow-hidden"
   modalHeading={$t('course.navItem.lessons.add_lesson.modal_heading')}
 >
-  <div
-    class="relative m-auto py-2 md:py-3 px-2 md:px-5 mb-2 md:mb-4 flex flex-wrap items-center dark:bg-neutral-800"
-  >
+
+<!-- used form to handle enter keydown event naturally -->
+<form on:submit={handleSubmit} class="relative m-auto py-2 md:py-3 px-2 md:px-5 mb-2 md:mb-4 flex flex-wrap items-center dark:bg-neutral-800 bg-yellow-500">
```
Comment (rotimi-best) [inline apps/dashboard/src/lib/components/Course/components/Lesson/NewLessonModal.svelte]: You can change this `on:submit={handleSubmit}` to `on:submit|preventDefault={handleSave}`

and then you don't need `handleSubmit` function

**Review by rotimi-best — CHANGES_REQUESTED** [review]
Thanks for the work on this @smritidoneria 🤩

Please attend to the following and I can merge it in.

**Comment by smritidoneria** [comment]
hey @rotimi-best , I have committed the changes as requested, kindly review it

File: apps/dashboard/src/lib/components/Course/components/Lesson/NewLessonModal.svelte [hunk 3]
```diff
@@ -44,19 +44,21 @@
 
     $handleAddLessonWidget.open = false;
   };
+
 </script>
 
 <Modal
+
```
Comment (rotimi-best) [inline apps/dashboard/src/lib/components/Course/components/Lesson/NewLessonModal.svelte]: Please remove this extra line

File: apps/dashboard/src/lib/components/Course/components/Lesson/NewLessonModal.svelte [hunk 4]
```diff
@@ -44,19 +44,21 @@
...
   width="w-[80%] md:w-[65%]"
   maxWidth="max-w-2xl"
   containerClass="overflow-hidden"
   modalHeading={$t('course.navItem.lessons.add_lesson.modal_heading')}
 >
-  <div
-    class="relative m-auto py-2 md:py-3 px-2 md:px-5 mb-2 md:mb-4 flex flex-wrap items-center dark:bg-neutral-800"
-  >
+
+<!-- used form to handle enter keydown event naturally -->
```
Comment (rotimi-best) [inline apps/dashboard/src/lib/components/Course/components/Lesson/NewLessonModal.svelte]: Please also remove this comment

File: apps/dashboard/src/lib/components/Course/components/Lesson/NewLessonModal.svelte [hunk 5]
```diff
@@ -93,9 +95,9 @@
...
     </div>
-  </div>
+  </form>
 
-  <div class="flex flex-row-reverse">
+  <div class="flex flex-row-reverse ">
     <PrimaryButton label={$t('course.navItem.lessons.add_lesson.save')} onClick={handleSave} />
   </div>
-</Modal>
+</Modal>
```
Comment (rotimi-best) [inline apps/dashboard/src/lib/components/Course/components/Lesson/NewLessonModal.svelte]: Please also restore the new line in this file.

**Comment by smritidoneria** [comment]
hey @rotimi-best , the changes are being made. kindly look at it

**Review by rotimi-best — APPROVED** [review]
Great job @smritidoneria 

Thank you

**Pull request merged**